paris-processor/src/main/java/com/airbnb/paris/processor/framework/SkyKotlinFile.kt

```diff
@@ -0,0 +1,45 @@
+package com.airbnb.paris.processor.framework
+
+import com.squareup.kotlinpoet.FileSpec
+import java.io.File
+import javax.annotation.processing.ProcessingEnvironment
+
+
+internal abstract class SkyKotlinFile(
+    private val packageName: String,
+    private val name: String,
+    private val block: FileSpec.Builder.() -> Unit
+) {
+
+    fun build(): FileSpec {
+        return FileSpec.builder(packageName, name).run {
+            block()
+            build()
+        }
+    }
+
+    /**
+     * See [kaptOutputPath]
+     */
+    fun write(outputPath: String) {
+        build().writeTo(File(outputPath))
```

Comment:
To simplify things I've been accessing the processor class instance directly to access utilities (`SkyExtensions.kt`) or even just things I don't want to have to pass around (like `RElement` in `ParisExtensions.kt`). It's a little unusual but in the case of annotation processors I think it can help simplify. What would you think of saving the kapt output path in a `SkyProcessor` field and accessing it directly? Given that we don't need to use different paths having it as a parameter seems unnecessary.

Reply:
sure